src/core/mempool.rs

```diff
@@ -1360,16 +1640,231 @@ impl MemPoolDB {
+                }
+            };
+            if contains {
+                // remote peer already has this one
+                continue;
+            }
+
+            let tx_bytes: Vec<u8> = row.get_unwrap("tx");
+            let tx = StacksTransaction::consensus_deserialize(&mut &tx_bytes[..])
+                .map_err(|_e| db_error::ParseError)?;
+
+            test_debug!("Returning txid {}", &txid);
+            ret.push(tx);
+            if (ret.len() as u64) >= max_txs {
+                break;
+            }
+        }
+
+        Ok(ret)
+    }
+
+    /// Stream transaction data
+    pub fn stream_txs<W: Write>(
+        &self,
+        fd: &mut W,
+        query: &mut TxStreamData,
+        count: u64,
+    ) -> Result<u64, ChainstateError> {
+        let mut num_written = 0;
+        while num_written < count {
+            if query.num_txs >= query.max_txs {
+                // don't serve more than this many txs
+                break;
+            }
+
+            // write out bufferred tx
+            while query.tx_buf.len() > 0 && query.tx_buf_ptr < query.tx_buf.len() {
+                let start = query.tx_buf_ptr;
+                let end = cmp::min(query.tx_buf.len(), ((start as u64) + count) as usize);
+                fd.write_all(&query.tx_buf[start..end])
+                    .map_err(ChainstateError::WriteError)?;
+
+                let nw = end.saturating_sub(start) as u64;
+                if nw == 0 {
+                    break;
+                }
+                query.tx_buf_ptr = end;
+                num_written += nw;
+            }
+
+            // load next
+            let mut next_txs = self.find_next_missing_transactions(
+                &query.tx_query,
+                query.height,
+                &query.last_txid,
+                1,
+                MAX_BLOOM_COUNTER_TXS.into(),
+            )?;
+            if let Some(next_tx) = next_txs.pop() {
+                query.tx_buf_ptr = 0;
+                query.tx_buf.clear();
+                query.num_txs += 1;
+
+                next_tx
+                    .consensus_serialize(&mut query.tx_buf)
+                    .map_err(ChainstateError::CodecError)?;
+
+                // find next page
+                let sql = "SELECT hashed_txid FROM randomized_txids WHERE txid = ?1 LIMIT 1";
+                let args: &[&dyn ToSql] = &[&next_tx.txid()];
+                let last_txid = match query_row(&self.conn(), sql, args)? {
+                    Some(txid) => txid,
+                    None => {
+                        // done!
+                        break;
+                    }
+                };
+                query.last_txid = last_txid;
+            } else {
+                // no more
+                break;
+            }
+        }
+        Ok(num_written)
+    }
 }
 
 #[cfg(test)]
 mod tests {
+    use std::cmp;
+    use std::collections::HashSet;
+    use std::io;
+
```

Comment:
Can you move this module to another file? After this PR, roughly 60% of this file will be tests, and 3.5k line files with mixtures of tests and code make the codebase much harder to navigate.

Reply:
Yup, done